### PR TITLE
SE blocking API for WebServer

### PR DESCRIPTION
**Pull request by tomas-langer**
Related to #2341 

This is a brand new feature to support blocking APIs that work nicely with project Loom.
Support for virtual threads is incorporated into the `ThreadPoolSupplier`, so it can be used in any part of Helidon simply by configuring `virtual-threads: true` for that executor.

This PR contains the API, implementation, and example. It is tested with native image.

This is first (experimental) approach, for discussion. Main features:

1. Blocking APIs do not introduce new routing abstraction
2. Main access point is `BlockingHandler` that allows you to create blocking routing methods and register them as usual with WebServer routing
3. All blocking calls are invoked in a custom executor service, that is by default using virtual threads when running on Loom enabled JVM
4. I have moved Loom support into `ThreadPoolSupplier`, so we can use it anywhere in Helidon (also refactored existing usage)

### CODE REVIEW

**Review by romain-grecourt — COMMENTED** [review]
- How about a new type of routing, it would make the example a little more elegant so that one does to explicitly use BlockingHandler.

- How about adding this the webserver module directly?

- There is a some duplication of the interfaces (BlockingRequest, BlockingResponse), we could probably extract the common stuff to compose the interfaces.

**Comment by tomas-langer** [comment]
I had a version with a dedicated routing type, but that could not be combined with the existing `Routing`. I like this approach, as I can chose for each route whether it is reactive or blocking.
Also the `BlockingHandler` was needed still, as there are too many possible method signatures - with reactive, it is just one `(ServerRequest, ServerResponse)`, with blocking it is:

1. `void (T entity)` for accepting an entity (PUT/POST)
2. `T (T entity)` for returning an entity with entity request
3. `T (BlockingRequest request)` for returning an entity and using request (e.g. to obtain headers, path params etc.)
4. `BlockingResponse (BlockingRequest)` for full control
5 `T ()` for a simple response

I may have missed some combinations here as well...

**Comment by tomas-langer** [comment]
Adding to webserver - yes, that may be useful, as the API is quite small. If we would end up with custom routing, I would rather have a separate module

Duplication - yes, that can be reduced (even though I did an implementation that is quite complete, this is still very much a POC and a topic for discussion)

**Comment by romain-grecourt** [comment]
Let's discuss this offline, I think having custom handle and routing are not exclusive.

File: examples/webserver/blocking/src/main/java/io/helidon/webserver/examples/blocking/Main.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                .config(config.get("server"))
+                .addMediaSupport(JsonpSupport.create())
+                .build();
+
+        server.start()
+                .thenAccept(ws -> {
+                    int backendPort;
+                    if (clientPort == -1) {
+                        backendPort = ws.port();
+                    } else {
```
Comment (danielkec) [inline examples/webserver/blocking/src/main/java/io/helidon/webserver/examples/blocking/Main.java:86]: I would expect something like 
```java
        server.start();
        System.out.println("WEB server is up! http://localhost:" + server.port());
        server.whenShutdown().thenRun(() -> {
            System.out.println("WEB server is DOWN. Good bye!");
        });
```

Comment (danielkec) [reply examples/webserver/blocking/src/main/java/io/helidon/webserver/examples/blocking/Main.java:86]: Maybe `server.startAndBlock()`

Comment (tomas-langer) [reply examples/webserver/blocking/src/main/java/io/helidon/webserver/examples/blocking/Main.java:86]: You can use `server.start().await()` - I think blocking API is mostly needed for routing, not so much for starting/stopping the webserver.

**Comment by tomas-langer** [comment]
Closing for now. Will revisit later

**Pull request closed without merging**